lms/djangoapps/bookmarks/api.py

```diff
@@ -90,4 +91,25 @@ def delete_bookmark(user, usage_key):
         ObjectDoesNotExist: If a bookmark with the parameters does not exist.
     """
     bookmark = Bookmark.objects.get(user=user, usage_key=usage_key)
+    _track_event('edx.course.bookmark.removed', user.username, usage_key)
     bookmark.delete()
+
+
+def _track_event(event_name, username, usage_key):
+    """
+    Track bookmark create/delete requests
+
+    Arguments:
+        event_name: name of event to track
+        username: username of requested user
+        usage_key: usage key of the XBlock
+    """
+    tracker.emit(
+        event_name,
+        {
+            'bookmark_id': '{},{}'.format(username, unicode(usage_key)),
```

Comment:
I think you should move the `resource_id` method from the serializer to the model and just pass `event_name` and the bookmark object to this method. Then you can pick all these properties from the object.